xamarin/Xamarin.Forms
# [iOS] Fixes crash in CollectionViews when using DataTemplateSelector as a DataTemplate when used in GroupHeaders and or GroupFooters

### Description of Change ###

CollectionViews with GroupHeaders and or GroupFooters will be able to use DataTemplateSelectors and DataTemplates. (Used to crash when you would set a Header/ Footer to a DataTemplateSelector)

CollectionViews now will only display A Group Header or Footer when IsGrouped is set to true AND a datatemplate is set for that Header and or Footer. (It used to display Header and Footer regardless of the fact that a DataTemplate was selected)

Some minor code cleanup.

### Issues Resolved ### 
- fixes #8647 
### API Changes ###
None

Added:
 None

### Platforms Affected ### 
- iOS

### Behavioral/Visual Changes ###
An iOS user will be able to use A DataTemplateSelector in a collectionview without crashing the App/

### Testing Procedure ###
Open ControlGalary App, select test 8647 and see that it doesn't crash, and only shows Group Headers (Not footers)

Video: (QuickTime)

[8647-TestRun.zip](https://github.com/xamarin/Xamarin.Forms/files/3892494/8647-TestRun.zip)



### PR Checklist ###
